File: project/jimmer-sql-test-model/src/main/java/org/babyfish/jimmer/sql/model/fetcher/Issue1434Message.java
package org.babyfish.jimmer.sql.model.fetcher;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.ForeignKeyType;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.JoinColumn;
import org.babyfish.jimmer.sql.ManyToOne;
import org.babyfish.jimmer.sql.Table;
import org.babyfish.jimmer.sql.Transient;
import org.jspecify.annotations.Nullable;

@Entity
@Table(name = "ISSUE_1434_MESSAGE")
public interface Issue1434Message {

    @Id
    long id();

    @Nullable
    @ManyToOne
    @JoinColumn(name = "USER_ID", foreignKeyType = ForeignKeyType.FAKE)
    Issue1434User user();

    @Transient(Issue1434MessageUserDepartmentNamesResolver.class)
    String userDepartmentNames();
}

File: project/jimmer-sql-kotlin-test-model/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/fetcher/Issue1434Message.kt
package org.babyfish.jimmer.sql.kt.model.fetcher

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.ForeignKeyType
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.JoinColumn
import org.babyfish.jimmer.sql.ManyToOne
import org.babyfish.jimmer.sql.Table
import org.babyfish.jimmer.sql.Transient

@Entity
@Table(name = "ISSUE_1434_MESSAGE")
interface Issue1434Message {

    @Id
    val id: Long

    @ManyToOne
    @JoinColumn(name = "USER_ID", foreignKeyType = ForeignKeyType.FAKE)
    val user: Issue1434User?

    @Transient(Issue1434MessageUserDepartmentNamesResolver::class)
    val userDepartmentNames: String
}
